src/TicTacToe.Services/GameService.cs

```diff
@@ -0,0 +1,162 @@
+                                          .ToList();
+
+            return games;
+        }
+
+        /// <inheritdoc />
+        public GameStatusOutput Create(GameCreationInput input, Guid creatorUserId)
+        {
+            var game = new Game()
+            {
+                Name = input.Name,
+                Visibility = input.Visibility,
+                HashedPassword = input.Password,
+                CreatorUserId = creatorUserId,
+                State = GameState.WaitingForASecondPlayer
+            };
+
+            context.Games.Add(game);
+            context.SaveChanges();
+
+            return game.ToGameStatusOutput();
+        }
+
+        /// <inheritdoc />
+        public GameStatusOutput Join(Guid gameId, Guid opponentUserId)
+        {
+            Random r = new Random();
+            var game = this.context.Games
+                .Include(g => g.CreatorUser)
+                .FirstOrDefault(g => g.GameId == gameId);
+
+            game.OpponentUserId = opponentUserId;
+            var turn = r.Next(0, 2);
+            if (turn == 0)
+            {
+                game.State = GameState.CreatorTurn;
+            }
+            else
+            {
+                game.State = GameState.OpponentTurn;
+            }
+
+            context.SaveChanges();
+
+            return game.ToGameStatusOutput();
+        }
+
+        /// <inheritdoc />
+        public GameStatusOutput Status(Guid gameId, Guid userId)
+        {
+            var game = this.context.Games.Include(x => x.CreatorUser)
+                                         .Include(x => x.OpponentUser)
+                                         .AsNoTracking()
+                                         .FirstOrDefault(g => g.GameId == gameId);
+
+            return game.ToGameStatusOutput();
+        }
+
+        /// <inheritdoc />
+        public GameStatusOutput Play(Guid gameId, Guid userId, int row, int col)
+        {
+            var game = this.context.Games.Include(x => x.CreatorUser)
+                .Include(x => x.OpponentUser)
+                .FirstOrDefault(g => g.GameId == gameId);
+
+            var user = this.context.Users.Include(u => u.Scores).AsNoTracking().FirstOrDefault(u => u.UserId == userId);
+
+            var boardArray = game.Board.ToCharArray();
+            var position = 3 * row + col;
+            char playerChar;
+
+            if (game.State == GameState.CreatorTurn)
+            {
+                playerChar = 'X';
+            }
+            else
+            {
+                playerChar = 'O';
+            }
+
+            boardArray[position] = playerChar;
+            string boardInsertedPosition = string.Join(string.Empty, boardArray);
+            var gameResult = gameValidator.GetGameResult(boardInsertedPosition);
+
+            game.Board = boardInsertedPosition;
+
+            if (gameResult == GameResult.NotFinished)
+            {
```

Comment:
Separate the if-else checks into a new method to improve the method readability and to minimize the complexity.